### DISCUSSION TITLE
Exchange matrix decomposition.

### DISCUSSION THREAD

**Discussion by adrybakov** [mailhexu/TB2J#41] [Q&A] [answered]
Hello!

Last year I've asked the [question](https://groups.google.com/g/tb2j/c/bBYrsS__zik) about the matrix decomposition into isotropic, anisotropic and dmi parts on the google forum. Lately I noticed that TB2J produces correctly decomposed matrices, however, there is nothing written about it in the [release notes](https://tb2j.readthedocs.io/en/latest/src/ReleaseNotes.html).

Is the problem resolved or am I just lucky with a particular system?

Best,
Andrey

**Comment by mailhexu** [comment, marked as answer]
I think it is just for the particular system. I am now working on an alternative way to compute/merge the DMI and anisotropic exchanges. Not sure if the anisotropic J will be traceless though. More tests needed. 
Best wishes, 
HeXu
